cql3/functions/user_function.hh

```diff
@@ -42,6 +43,10 @@ private:
     bool _called_on_null_input;
     context _ctx;
 
+    // wasm UDF instance cache, allowing maximum concurrency
+    std::queue<wasmtime::Store> _store;
+    std::queue<wasmtime::Instance> _instance;
+    std::queue<wasmtime::Func> _func;
```

Comment:
I think that instead of 3 queues we just need a single queue which keeps a struct. And this struct should wrap _store, _instance and _func. Am I correct that we always use all three? In this case it makes perfect sense to only use a single queue. That would make the call site look nicer as well.

Reply:
The cache has to be shard-global, otherwise many distinct functions can eat all shard memory.